### DISCUSSION TITLE
How to connect to remote language serververs

### DISCUSSION THREAD

**Discussion by QuinnFreedman** [helix-editor/helix#5394] [Q&A] [answered]
I have a python environment running inside a docker container. `pylsp` is installed in the container, along with all the other plugins and packages. The local project folder is bound to a volume inside the container, so I still want to edit the files locally, but I want to talk to the language server in the container.

There are two ways I could think of to do this:
1. Set the `language-server.command` in `languages.toml` to something like `docker run -it --rm app pylsp`. This will start the docker container and pipe stdin/stdout to pylsp. This almost works, but there is an issue with file paths. I get the error: `mypy: can't read file '/home/myusername/myproject/filename.py': No such file or directory`. That makes sense because the docker container can't see my `/home`. It has the `myproject` folder mounted as a volume. Is there a way to specify a regex or something to translate `/home/myusername/myproject/filename.py` to `/code/filename.py` when passing the file to the LSP? or to use relative paths somehow?
2. pylsp (and most LSPs) support communication over websockets or tcp instead of stdio. Does helix support connecting to a remote language server?

I have searched the documentation and discussions for this but havent been able to find anything. Please let me know if anyone has any ideas!

**Comment by kirawi** [comment, marked as answer]
Regarding method 1: https://github.com/helix-editor/helix/pull/5204

**Reply by QuinnFreedman** [reply]
Thanks, that looks like exactly what I was looking for. I hope that gets merged soon

**Reply by a-berg** [reply]
soon being never, yeah

**Comment by larsblumberg** [comment]
Until helix doesn't support path remapping, one can follow my blog post at https://nlited.de/using-helix-editor-with-a-dockerized-python-lsp which describes how to connect a dockerized Python LSP to helix, tackling the path remapping problem via docker volume mapping.
